src/editor/bin/Input.re

```diff
@@ -32,58 +37,68 @@ let charToCommand = (codepoint: int, mods: Modifier.t) => {
 
   let key =
     keyPressToString(~shiftKey=false, ~altKey, ~ctrlKey, ~superKey, char);
-  Some(key);
+  let shouldOniListen = isOniModifier(~altKey, ~ctrlKey, ~superKey, ~key);
+  Some((key, shouldOniListen));
 };
 
 let keyPressToCommand =
-    (key: Key.t, buttonState: ButtonState.t, mods: Modifier.t) =>
-  if (buttonState == GLFW_PRESS || buttonState == GLFW_REPEAT) {
-    /* If ctrl is pressed, it's a non printable character, not handled by charMods - so we handle any character */
-    let key =
-      if (Modifier.isControlPressed(mods)) {
+    ({shiftKey, altKey, ctrlKey, superKey, key, _}: Events.keyEvent) => {
+  let keyString =
+    ctrlKey
+      ? /**
+        TODO:
+        Revery's toString method returns lower case
+        characters which need to be capitalized. Instead we
+        should use ?derving show (we will need to format out the KEY_ prefix)
+        or convert the return values to uppercase
+       */
+        Some(Revery.Key.toString(key) |> String.capitalize_ascii)
+      : (
         switch (key) {
-        | _ => Some(Key.show(key))
-        };
-      } else {
-        switch (key) {
-        | GLFW_KEY_ESCAPE => Some("ESC")
-        | GLFW_KEY_TAB => Some("TAB")
-        | GLFW_KEY_ENTER => Some("CR")
-        | GLFW_KEY_BACKSPACE => Some("BS")
-        | GLFW_KEY_LEFT => Some("LEFT")
-        | GLFW_KEY_RIGHT => Some("RIGHT")
-        | GLFW_KEY_DOWN => Some("DOWN")
-        | GLFW_KEY_UP => Some("UP")
-        | GLFW_KEY_LEFT_SHIFT
-        | GLFW_KEY_RIGHT_SHIFT => Some("SHIFT")
+        | KEY_ESCAPE => Some("ESC")
+        | KEY_TAB => Some("TAB")
+        | KEY_ENTER => Some("CR")
+        | KEY_BACKSPACE => Some("BS")
+        | KEY_LEFT => Some("LEFT")
+        | KEY_RIGHT => Some("RIGHT")
+        | KEY_DOWN => Some("DOWN")
+        | KEY_UP => Some("UP")
+        | KEY_LEFT_SHIFT
+        | KEY_RIGHT_SHIFT => Some("SHIFT")
         | _ => None
-        };
-      };
+        }
+      );
+  switch (keyString) {
+  | None => None
+  | Some(k) =>
+    let keyPressString =
+      keyPressToString(~shiftKey, ~altKey, ~ctrlKey, ~superKey, k);
+    let shouldOniListen =
+      isOniModifier(~altKey, ~ctrlKey, ~superKey, ~key=keyPressString);
 
-    switch (key) {
-    | None => None
-    | Some(v) =>
-      let altKey = Modifier.isAltPressed(mods);
-      let ctrlKey = Modifier.isControlPressed(mods);
-      let superKey = Modifier.isSuperPressed(mods);
-      let shiftKey = Modifier.isShiftPressed(mods);
-      let keyToSend =
-        keyPressToString(~shiftKey, ~altKey, ~ctrlKey, ~superKey, v);
-      Some(keyToSend);
-    };
-  } else {
-    None;
+    Some((keyPressString, shouldOniListen));
   };
+};
+
+/**
+   Search if any of the matching "when" conditions in the Keybindings.json
```

Comment:
Cool! Thanks for implementing these modes, and matching them up with VSCode's modes 👍